orderBook.js

```diff
@@ -0,0 +1,34 @@
+function reconcileOrder(existingBook, incomingOrder) {
+  const oppOrderTypes = existingBook.filter(order => order.type !== incomingOrder.type)
+  let updatedBook = existingBook.filter(order => order.type === incomingOrder.type)
+  updatedBook = oppOrderTypes.length
+    ? updatedBook.concat(determineSamePrice(oppOrderTypes, incomingOrder))
+    : updatedBook.concat(incomingOrder)
+  return updatedBook
+}
```

Comment:
If you dealt with one order at a time you wouldn't need so many loops. You're filtering twice here and then later doing a findIndex. See the solution branches.